### PR TITLE
Fix L026 false positive on "SELECT INTO" statement

### LINKED ISSUE #1830: Exasol L026 - Fails on SELECT INTO TABLE statement
In Exasol we could insert data into a table like this:

```
select * into table schm.tab from schm.tab2;
```

## Expected Behaviour
Rule L026 should pass

## Observed Behaviour
Rule was raised
```
L:   2 | P:  21 | L026 | Reference 'schm.tab' refers to table/view not found in
                       | the FROM clause or found in parent subquery
```

## Steps to Reproduce
Lint the statement above

## Dialect
exasol

## Version
Include the output of `sqlfluff --version` along with your Python version
sqlfluff 0.7.1
python 3.8.12
## Configuration
default

### PR DESCRIPTION
**Pull request by barrywhart**
### Brief summary of the change made
Fixes #1830

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/rules/L026.py [hunk 1]
```diff
@@ -122,19 +122,33 @@ def _analyze_table_references(
                 # Try and resolve each reference to a value in query.aliases (or
                 # in an ancestor query).
                 for r in select_info.reference_buffer:
-                    # This function walks up the query's parent stack if necessary.
-                    violation = self._resolve_reference(
-                        r, self._get_table_refs(r, dialect), dml_target_table, query
-                    )
-                    if violation:
-                        violations.append(violation)
+                    if not self._should_ignore_reference(r, selectable):
```
Comment (barrywhart) [inline src/sqlfluff/rules/L026.py:125]: Lines 126-131 are not changed; I simply moved them inside a new `if` statement.

File: src/sqlfluff/rules/L026.py [hunk 2]
```diff
@@ -122,19 +122,33 @@ def _analyze_table_references(
...
 
         # Visit children.
         for child in query.children:
             self._analyze_table_references(
                 cast(L026Query, child), dml_target_table, dialect, violations
             )
 
+    @staticmethod
+    def _should_ignore_reference(reference, selectable):
+        if selectable.dialect.name == "exasol":
```
Comment (tunetheweb) [inline src/sqlfluff/rules/L026.py:141]: Why just exasol? Surely this is common to many (all?) dialects?

T-SQL has this for example.

The other dialects SHOULD have type set up IMHO this but don't. Whether we want to change that as part of this PR, or just leave that for later is up to you, but either way I think we should remove this `if` statement.

**Review by tunetheweb — CHANGES_REQUESTED** [review]

File: src/sqlfluff/rules/L026.py (hunk 2 above)
Comment (jpy-git) [reply src/sqlfluff/rules/L026.py:141]: Yeah I think removing it makes sense

Comment (barrywhart) [reply src/sqlfluff/rules/L026.py:141]: Good catch! `SELECT INTO` sounded like it _might_ be generic, but I didn't think to check.

I'll remove the `if` statement, but would rather not make any dialect changes in this PR (scope creep). Want to create an issue for that, or should we wait until a user reports it?

Comment (tunetheweb) [reply src/sqlfluff/rules/L026.py]: Happy to wait for reports.

L026 is disabled by default for some dialects anyway ("bigquery", "hive", "redshift").

**Pull request merged**